### PR TITLE
[stable12] Send an email once a file/folder is shared with a user

### PR DESCRIPTION
**Pull request by MorrisJobke**
* backport of #5897

### CODE REVIEW

**Review by LukasReschke — APPROVED** [review]

File: lib/private/Share20/Manager.php [hunk 1]
```diff
@@ -666,10 +684,91 @@ public function createShare(\OCP\Share\IShare $share) {
...
+	 * @param string $initiator user ID of share sender
+	 * @param string $shareWith email address of share receiver
+	 * @throws \Exception If mail couldn't be sent
+	 */
+	protected function sendMailNotification($filename,
+											$link,
+											$initiator,
+											$shareWith) {
+		$initiatorUser = $this->userManager->get($initiator);
+		$initiatorDisplayName = ($initiatorUser instanceof IUser) ? $initiatorUser->getDisplayName() : $initiator;
```
Comment (LukasReschke) [inline lib/private/Share20/Manager.php]: This will never work. Also please add tests covering this case. THX!

Comment (LukasReschke) [inline lib/private/Share20/Manager.php]: To be more specific: IUser doesn't exist here

**Review by LukasReschke — DISMISSED** [review]

File: lib/private/Share20/Manager.php (hunk 1 above)
Comment (MorrisJobke) [inline lib/private/Share20/Manager.php]: Fixed

**Comment by MorrisJobke** [comment]
@nickvergessen @rullzer @LukasReschke @schiessle Please review

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**